pkg/analyzer/testdata/src/defer_in_loop/inline_funcs.go

```diff
@@ -0,0 +1,24 @@
+package defer_in_loop
+
+func inlineFuncs() {
+	for i := 0; i < 10; i++ {
+		// inline function declaration and call
+		_ = func() error {
+			defer func() { // this defer is in a different function, not in a loop per-se
+			}()
+			return nil
+		}()
+
+		// inline function declaration and call with "go"
+		go func() {
+			defer func() { // this defer is in a different function, not in the loop per-se
+			}()
+		}()
+
+		// inline function declaration without calling it
+		_ = func() {
+			defer func() { // this defer is in a different function, not in the loop per-se
+			}()
+		}
```

Comment:
would be nice to have a test case that we DO get a warning if we put a defer after the excluded cases.
Also, something like "a defer inside a loop inside a function closure inside a loop" which should get a warning, but is a bit deeper?